server/models/db/availability.js

```diff
@@ -0,0 +1,38 @@
+const { db } = require('../../lib/connection');
+
+//const getAvailabilityDetails = require('../data/AvailabilityDetails');
+
+const Tables = require('../../constants/tables.json');
+
+//Parameters returned are slot_id, school_id, worker_id, status, and date
+async function getAvailabilityDetails( workerId, schoolId, startTime, endTime) {
+    try{
+    //var queryParams = `${Tables.Worker_Timeslot}.slot_id, ${Tables.workerTimeslot}.school_id, ${Tables.timeslot}.worker_id, ${Tables.timeslot}.date, ${Tables.Worker_Timeslot}.status`;
+    //var queryCond = `${Tables.Worker_Timeslot}.worker_id='${worker_id} and ${Tables.Worker_Timeslot}.school_id='${school_id} and ${Tables.Worker_Timeslot}.worker_id='${worker_id}'`
+
+    // Execute query to obtain all need Worker_Timeslot information for a worker.
+
+    //queryParams += `, ${Tables.Worker_Timeslot}.slot_id, ${Tables.workerTimeslot}.school_id, ${Tables.timeslot}.worker_id, ${Tables.timeslot}.date, ${Tables.Worker_Timeslot}.status`;
+    //queryCond += ` and ${Tables.Worker_Timeslot}.worker_id='${worker_id}'`;
+
+    //The queryStatement selects the parameters: slot_id, school_id, worker_id, status, and date
+    // from the Worker_Timeslot table
+    //where worker_id = the given worker_id
+
+    const queryStatement = `select * from worker_timeslot natural join timeslot where school_id = ${schoolId} and worker_id = ${workerId} and start_time = '${startTime}' and end_time = '${endTime}' ;`;
```

Comment:
Hi Nancy,
Currently, this query returns ALL the worker timeslot records. However, we only want the available times for the service worker. Thus, I would recommend adding another where condition to check that the status does indeed match `available` (and I would also suggest leveraging the `timeslotStatus.json` file in the `constants` folder to match the string.

Reply:
Hi Amy,
Thank you for suggesting that!

My action: I've added a where clause to the `queryStatement` such that:
the `queryStatement` returns values where status obtained from the timeslotStatus.json table = ${'available'}.

You can now see this change in line 22 of server/models/db/availability.js

Please confirm if this takes care of your suggestion!